src/main/java/com/aws/greengrass/clientdevices/auth/api/ClientDevicesAuthServiceApi.java

```diff
@@ -51,21 +51,26 @@ public ClientDevicesAuthServiceApi(SessionManager sessionManager, DeviceAuthClie
      * @param certificatePem PEM encoded client certificate.
      * @return True if the provided client certificate is trusted.
      */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
     public boolean verifyClientDeviceIdentity(String certificatePem) {
-        boolean success;
-
-        // Allow internal clients to verify their identities
-        if (deviceAuthClient.isGreengrassComponent(certificatePem)) {
-            success = true;
-        } else {
-            success = useCases.get(VerifyIotCertificate.class).apply(certificatePem);
+        try {
+            boolean success;
+
+            // Allow internal clients to verify their identities
+            if (deviceAuthClient.isGreengrassComponent(certificatePem)) {
+                success = true;
+            } else {
+                success = useCases.get(VerifyIotCertificate.class).apply(certificatePem);
+            }
+
+            domainEvents.emit(new VerifyClientDeviceIdentityEvent(success ? VerifyClientDeviceIdentityEvent
+                    .VerificationStatus.SUCCESS : VerifyClientDeviceIdentityEvent.VerificationStatus.FAIL));
+
+            return success;
+        } catch (RuntimeException e) {
+            domainEvents.emit(new ServiceErrorEvent());
+            throw e;
```

Comment:
is this being logged anywhere? We're going to have a very hard time investigating errors if they're never being logged.

Reply:
It's logged in the IPC handler, but since those aren't called in every instance another approach I had was to log in the event handler for each service error and remove the other logs to avoid duplicates

Reply:
I had an approach earlier to log each service error event in the handler with the exception and an error message. Removed it after this conversation: [https://github.com/aws-greengrass/aws-greengrass-client-device-auth/pull/208#discussion_r1046070557](url)

I think now it would be good to add it back since the existing logs don't apply to runtime exceptions

Reply:
it's being logged at the ipc level, but forgot that this could be called directly--i take back my original comment :)

Reply:
Yup I'd forgotten that too. Changed it to just log when the exception happens instead of having to go through the handler this time